kernel/src/snapshot.rs

```diff
@@ -366,7 +366,7 @@ impl Snapshot {
 }
 
 // Note: Schema can not be derived because the checkpoint schema is only known at runtime.
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
```

Comment:
qq: What does it mean for two `Schema`s to be equal? Is metadata included in this?